### PR TITLE
Updated objects explainer to reflect current state

### PR DESCRIPTION
**Pull request by pipobscure**

### CODE REVIEW

**Comment by pipobscure** [comment]
Merging upon advice of @littledan

File: objects.md [hunk 1]
```diff
@@ -205,12 +216,15 @@ of `Instant` and the second is a `string` representing a valid IANA timezone.
 
 #### ZonedDateTime.prototype
 
-`ZonedDateTime.prototype` includes all properties and methods of `OffsetDateTime.prototype`, plus the following additions and replacements.
+`ZonedDateTime.prototype` includes all properties and methods of `OffsetDateTime.prototype` except `plus` & `minus`.
```
Comment (gibson042) [inline objects.md]: Doesn't that destroy the whole idea of a [class hierarchy](https://github.com/tc39/proposal-temporal/issues/120#user-content-class-relationships)? Why should ZonedDateTime not support the same arithmetic as OffsetDateTime?

File: objects.md [hunk 2]
```diff
@@ -166,10 +168,21 @@ The algorithm is such that:
 
 #### OffsetDateTime.prototype.minus(other: [ZonedDateTime](#ZonedDateTime)): [Duration](#Duration)
```
Comment (gibson042) [inline objects.md]: `other: [ZonedDateTime](#ZonedDateTime)` → `value: [DurationLike](#DurationLike)`

File: objects.md [hunk 3]
```diff
@@ -290,19 +304,32 @@ The `.weekOfYear` property represents the ISO week-number. Beware that dates at
...
 
 The algorithm is such that:
   1. the individual values are added to the existing values.
   2. the range of `days` is ensured to be between 1 and 29-31 depending on the month by adjusting `month`
   3. the range of `months` is ensured to be between 1 and 12 by adjusting the `years`.
 
+#### CivilDate.prototype.plus(value: [DurationLike](#DurationLike)) : [CivilDate](#CivilDate)
+
+Creates a new `CivilDate` object by subtracting values to its members.
+The specified values must be numeric if specified. The specified values
```
Comment (gibson042) [inline objects.md]: The first sentence on this line is redundant with its successor.

File: objects.md [hunk 4]
```diff
@@ -506,10 +500,12 @@ The `.microsecond` property represents the sub-millisecond component of the mill
 
 The `.nanosecond` property represents the sub-microsecond component of the microsecond of the `CivilDateTime` with nanosecond precision. It will have a value between 0 and 999.
 
-#### CivilDateTime.prototype.plus(inteval: [DurationLike](#DurationLike)) : [CivilTime](#CivilTime)
+#### CivilDateTime.prototype.plus(duration: [DurationLike](#DurationLike)) : [CivilDateTime](#CivilDateTime)
 
-Creates a new `CivilDateTime` object by adding (subtracting for negative values) values to its members.
-The specified values must be numeric if specified.
+Creates a new `CivilDateTime` object by subtracting values to its members.
```
Comment (gibson042) [inline objects.md]: subtracting → adding

File: objects.md [hunk 5]
```diff
@@ -290,19 +304,32 @@ The `.weekOfYear` property represents the ISO week-number. Beware that dates at
 
 #### CivilDate.prototype.plus(value: [DurationLike](#DurationLike)) : [CivilDate](#CivilDate)
 
-Creates a new `CivilDate` object by adding (subtracting for negative values) values to its members.
-The specified values must be numeric if specified.
+Creates a new `CivilDate` object by adding values to its members.
+The specified values must be numeric if specified. The specified values
```
Comment (gibson042) [inline objects.md]: The first sentence on this line is redundant with its successor.

File: objects.md [hunk 6]
```diff
@@ -398,11 +387,18 @@ The `.microsecond` property represents the sub-millisecond component of the mill
...
+#### CivilTime.prototype.with(values: [TimeLike](#TimeLike)) : [CivilTime](#CivilTime)
 
-Creates a new `CivilTime` object by adding (subtracting for negative values) values to its members.
+Creates a new `CivilTime` object by overriding specified values to its members.
 The specified values must be numeric if specified.
 
+#### CivilTime.prototype.plus(values: [DurationLike](#DurationLike)) : [CivilTime](#CivilTime)
+
+Creates a new `CivilTime` object by subtracting values to its members.
+The specified values must be numeric if specified. The specified values
```
Comment (gibson042) [inline objects.md]: The first sentence on this line is redundant with its successor.

File: objects.md [hunk 7]
```diff
@@ -506,10 +500,12 @@ The `.microsecond` property represents the sub-millisecond component of the mill
 
 The `.nanosecond` property represents the sub-microsecond component of the microsecond of the `CivilDateTime` with nanosecond precision. It will have a value between 0 and 999.
 
-#### CivilDateTime.prototype.plus(inteval: [DurationLike](#DurationLike)) : [CivilTime](#CivilTime)
+#### CivilDateTime.prototype.plus(duration: [DurationLike](#DurationLike)) : [CivilDateTime](#CivilDateTime)
 
-Creates a new `CivilDateTime` object by adding (subtracting for negative values) values to its members.
-The specified values must be numeric if specified.
+Creates a new `CivilDateTime` object by subtracting values to its members.
+The specified values must be numeric if specified. The specified values
```
Comment (gibson042) [inline objects.md]: The first sentence on this line is redundant with its successor.

File: objects.md [hunk 8]
```diff
@@ -519,12 +515,24 @@ The algorithm is such that:
  5. the range of `seconds` is ensured to be between 0 and 59 by adjusting the `minutes`
  6. the range of `minutes` is ensured to be between 0 and 59 by adjusting the `hours`
  7. the range of `hours` is ensured to be between 0 and 23
+ 8. the range of `days` is ensured to be between 1 and 29-31 depending on the month by adjusting `month`
+ 9. the range of `months` is ensured to be between 1 and 12 by adjusting the `years`.
```
Comment (gibson042) [inline objects.md]: I think the order of items in these lists is backwards (cf. #58).

Comment (gibson042) [inline objects.md]: I think the order of items in these lists is backwards (cf. #58).

File: objects.md [hunk 9]
```diff
@@ -519,12 +515,24 @@ The algorithm is such that:
...
 
+#### CivilDateTime.prototype.minus(duration: [DurationLike](#DurationLike)) : [CivilDateTime](#CivilDateTime)
+#### CivilDateTime.prototype.difference(other: TimeLike) : [Duration](#Duration)
 #### CivilDateTime.prototype.with(values: [DateTimeLike](#DateTimeLike)) : [CivilTime](#CivilTime)
 
 Creates a new `CivilTime` object by overriding specified values to its members.
 The specified values must be numeric if specified.
 
+#### CivilDateTime.prototype.withZone(ianaZone : string, filter?: string | symbol) : [ZonedDateTime](#ZonedDateTime)
```
Comment (gibson042) [inline objects.md]: What is `filter`?

Comment (gibson042) [inline objects.md]: What is `filter`?

File: objects.md [hunk 10]
```diff
@@ -632,30 +642,6 @@ Is the **integer** number of microseconds difference this interval represents.
...
 
-#### Duration.prototype.minus(other: Duration) : Duration
-
-Creates a new `Duration` where each of the fields is defined as:
-
- `field = mine.field - other.field`
-
-This can result in mixed intervals where some fields are positive and some negative. Such intervals are fully valid.
-
-#### Duration.prototype.plus(values: [DurationLike](#DurationLike)) : [Duration](#Duration)
```
Comment (gibson042) [inline objects.md]: Why is Duration arithmetic being removed?

**Pull request merged**